### PR TITLE
refactor(ui5-multi-combobox): introduce nested grouping of items

### PR DESCRIPTION
**Pull request by ivoplashkov**
Removes the `ui5-mcb-group-item` component.

BREAKING CHANGE: The `ui5-mcb-group-item` component is removed. Groups can now be created with the `ui5-mcb-group`. Instead of using `ui5-mcb-group-item` as separator in a flat structure:
```html
<ui5-multi-combobox placeholder="Select a country">
	<ui5-mcb-group-item text="Asia"></ui5-mcb-group-item>
        <ui5-mcb-item text="Afghanistan"></ui5-mcb-item>
        <ui5-mcb-item text="China"></ui5-mcb-item>
        <ui5-mcb-item text="India"></ui5-mcb-item>
        <ui5-mcb-item text="Indonesia"></ui5-mcb-item>
	<ui5-mcb-group-item text="Europe"></ui5-mcb-group-item>
        <ui5-mcb-item text="Austria"></ui5-mcb-item>
        <ui5-mcb-item text="Bulgaria"></ui5-mcb-item>
        <ui5-mcb-item text="Germany"></ui5-mcb-item>
        <ui5-mcb-item text="Italy"></ui5-mcb-item>
</ui5-multi-combobox>
```
The API supports nesting of `ui5-mcb-item` components inside an `ui5-mcb-item-group`:
```html
<ui5-multi-combobox placeholder="Select a country">
	<ui5-mcb-item-group text="Asia">
		<ui5-mcb-item text="Afghanistan"></ui5-mcb-item>
		<ui5-mcb-item text="China"></ui5-mcb-item>
		<ui5-mcb-item text="India"></ui5-mcb-item>
		<ui5-mcb-item text="Indonesia"></ui5-mcb-item>
	</ui5-mcb-item-group>
	<ui5-mcb-item-group text="Europe">
		<ui5-mcb-item text="Austria"></ui5-mcb-item>
		<ui5-mcb-item text="Bulgaria"></ui5-mcb-item>
		<ui5-mcb-item text="Germany"></ui5-mcb-item>
		<ui5-mcb-item text="Italy"></ui5-mcb-item>
	</ui5-mcb-item-group>
</ui5-multi-combobox>
```

Relates to https://github.com/SAP/ui5-webcomponents/issues/8461

### CODE REVIEW

File: packages/main/src/MultiComboBox.ts [hunk 1]
```diff
@@ -1361,9 +1364,25 @@ class MultiComboBox extends UI5Element implements IFormInputElement {
 		this.fireEvent("open-change");
 	}
 
+	getItems(): Array<IMultiComboBoxItem> {
```
Comment (ndeshev) [inline packages/main/src/MultiComboBox.ts:1367]: add a bit of documentation for this method

File: packages/website/docs/_components_pages/main/MultiComboBox/MultiComboBoxGroupItem.mdx [hunk 2]
```diff
@@ -1,5 +1,5 @@
 ---
-slug: ../../MultiComboBoxGroupItem
+slug: ../../MultiComboBoxGroup
```
Comment (ndeshev) [inline packages/website/docs/_components_pages/main/MultiComboBox/MultiComboBoxGroupItem.mdx:2]: MultiComboBoxItemGroup

File: packages/main/README.md [hunk 3]
```diff
@@ -50,7 +50,7 @@ Provides general purpose UI building blocks such as buttons, labels, inputs and
 | Message Strip            | `ui5-message-strip`         | `import "@ui5/webcomponents/dist/MessageStrip.js";`        |
 | Multi ComboBox           | `ui5-multi-combobox`        | `import "@ui5/webcomponents/dist/MultiComboBox.js";`       |
 | Multi ComboBox Item      | `ui5-mcb-item`              | comes with `ui5-multi-combobox`                            |
-| Multi ComboBox Group Item| `ui5-mcb-group-item`        | comes with `ui5-multi-combobox`                            |
+| Multi ComboBox Group     | `ui5-mcb-group`             | comes with `ui5-multi-combobox`                            |
```
Comment (ndeshev) [inline packages/main/README.md:53]: Multi ComboBox Item Group

**Review by ndeshev — APPROVED** [review]

**Review by elenastoyanovaa — CHANGES_REQUESTED** [review]
When you press F4 on a MCB with grouping the focus stays in the input, which interrups the screen reader modes. 
Other that that the speech output is as expected.

**Pull request merged**